dev/builddeps-veloxbe.sh

```diff
@@ -173,6 +173,8 @@ if [ "$ENABLE_VCPKG" = "ON" ]; then
     # vcpkg will install static depends and init build environment
     envs="$("$GLUTEN_DIR/dev/vcpkg/init.sh")"
     eval "$envs"
+else
+    echo "Dynamic linked libs based building is deprecated, please enabled VCPKG based building."
```

Comment:
can we reserve dynamic link? I found static build is unstable in CI, it's better keep both way.

Reply:
The dynamic link will be kept but we need to print one message. The static packaging has been used starts from gluten 1.0 and seems working well
The issue on dynamic packaging is on the dependencies update case(especially for the minor version update on libraries) , we need to update the related loader as well

https://github.com/apache/incubator-gluten/blob/main/backends-velox/src/main/scala/org/apache/gluten/utils/SharedLibraryLoaderUbuntu2204.scala

Reply:
https://github.com/apache/incubator-gluten/pull/7115#issuecomment-2330591059
https://github.com/apache/incubator-gluten/actions/runs/10769801803/job/29861861312

I found such errors, are these known issues?

Reply:
In fact, most of our customers have already use static link in their production ready environment.
One of the major reason is our 3rd party package script is no longer to be updated.
Only a few customers are still using dynamic link since they maintain their own 3rd party libraries instead to use the script in Gluten.

Reply:
> #7115 (comment)

Is the issue caused by static link?